docs/source/tutorial/modules.rst

```diff
@@ -129,6 +129,19 @@ this effectively makes the function definition part of the module's API.
 Therefore, it's generally a good idea to avoid using ``public export`` for
 functions unless you really mean to export the full definition.
 
+.. note::
+    *For beginners*: 
+    If the function needs to be accessed only at runtime, use ``export``.
+    However, if it's also meant to be used at *compile* time (e.g. to prove 
+    a theorem), use ``public export``.  
+    For example, consider the function ``plus: Nat -> Nat -> Nat`` discussed
+    previously, and the following theorem: ``thm: plus Z m = m``.  
+    In order to to prove it, the type checker needs to reduce ``plus Z m`` to ``m`` 
+    (and hence obtain ``thm: m = m``).
```

Comment:
Just a nitpick but I think it's common to have spaces around the `:` in type signatures.

```suggestion
    For example, consider the function ``plus : Nat -> Nat -> Nat`` discussed
    previously, and the following theorem: ``thm : plus Z m = m``.  
    In order to to prove it, the type checker needs to reduce ``plus Z m`` to ``m`` 
    (and hence obtain ``thm : m = m``).
```